test/async-hooks/test-ttywrap.readstream.js

```diff
@@ -1,32 +1,42 @@
 'use strict';
+
 const common = require('../common');
 const assert = require('assert');
-
-// general hook test setup
 const tick = require('./tick');
 const initHooks = require('./init-hooks');
 const { checkInvocations } = require('./hook-checks');
+const tty_fd = common.getTTYfd();
+
+if (tty_fd < 0)
+  return common.skip('no valid TTY fd available');
+const ttyStream = (() => {
+  try {
+    return new (require('tty').ReadStream)(tty_fd);
```

Comment:
So if we don't fear side effect, why not move the `require` to the top?
The invoked lambda I understand (`return null` and assign to const is very function-programing, and I like 👍 )